src/tini.c

```diff
@@ -133,8 +133,19 @@ int spawn(const signal_configuration_t* const sigconf_ptr, char* const argv[], i
 		}
 
 		execvp(argv[0], argv);
-		PRINT_FATAL("Executing child process '%s' failed: '%s'", argv[0], strerror(errno));
-		return 1;
+	    	// execvp will only return on an error so make sure that we check the errno
+		// and exit with the correct return status for the error that we encountered
+		int status = EXIT_FAILURE;
+		switch errno {
+		case ENOENT:
+			status = 127;
+			break;
+		case EACCES:
+			status = 126;
+			break;
+		}
+		fprintf(stderr, "%s\n", strerror(errno));
```

Comment:
That's a very valid point; I completely understand the desire for a more white-label approach on your end here (in fact that's why I brought it up!). 

So, how about:

- (Optional) We rename the NO_ARGS compile flag to DOCKERINIT (I'll make the change in Tini but you'll want to change that on your end as well).
- I update the macros so that the prefix is only showed when said flag is not set.
- You glance at the other warning and fatal messages (these are the only ones that are shown at the default verbosity, although we *could* quiet those) to check that everything looks reasonable. The errors are a little inconsistent as of now, although most of them are also very unlikely to come up.

For obvious reasons (mainly to avoid breaking things inadvertently!), I'd like to avoid having too many differences between the "docker version" and the "regular version" of Tini, but as long as we're only touching input (args) and output (log messages), I think we're fine.

Cheers,